pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -334,30 +334,38 @@ func (c *Reconciler) isAffinityAssistantDisabled(ctx context.Context) bool {
 }
 
 // getAssistantAffinityMergedWithPodTemplateAffinity return the affinity that merged with PipelineRun PodTemplate affinity.
-func getAssistantAffinityMergedWithPodTemplateAffinity(pr *v1.PipelineRun) *corev1.Affinity {
-	// use podAntiAffinity to repel other affinity assistants
-	repelOtherAffinityAssistantsPodAffinityTerm := corev1.WeightedPodAffinityTerm{
-		Weight: 100,
-		PodAffinityTerm: corev1.PodAffinityTerm{
-			LabelSelector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					workspace.LabelComponent: workspace.ComponentNameAffinityAssistant,
-				},
-			},
-			TopologyKey: "kubernetes.io/hostname",
-		},
-	}
-
+func getAssistantAffinityMergedWithPodTemplateAffinity(pr *v1.PipelineRun, aaBehavior aa.AffinityAssitantBehavior) *corev1.Affinity {
 	affinityAssistantsAffinity := &corev1.Affinity{}
 	if pr.Spec.TaskRunTemplate.PodTemplate != nil && pr.Spec.TaskRunTemplate.PodTemplate.Affinity != nil {
 		affinityAssistantsAffinity = pr.Spec.TaskRunTemplate.PodTemplate.Affinity
 	}
 	if affinityAssistantsAffinity.PodAntiAffinity == nil {
 		affinityAssistantsAffinity.PodAntiAffinity = &corev1.PodAntiAffinity{}
 	}
-	affinityAssistantsAffinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution =
-		append(affinityAssistantsAffinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution,
-			repelOtherAffinityAssistantsPodAffinityTerm)
+
+	RepelOtherAffinityAssistantsPodAffinityTerm := corev1.PodAffinityTerm{
+		LabelSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				workspace.LabelComponent: workspace.ComponentNameAffinityAssistant,
+			},
+		},
+		TopologyKey: "kubernetes.io/hostname",
+	}
+
+	// use podAntiAffinity to repel other affinity assistants
+	if aaBehavior == aa.AffinityAssistantPerPipelineRunWithIsolation {
+		affinityAssistantsAffinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution =
+			append(affinityAssistantsAffinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution,
+				RepelOtherAffinityAssistantsPodAffinityTerm)
+	} else {
+		preferredRepelOtherAffinityAssistantsPodAffinityTerm := corev1.WeightedPodAffinityTerm{
```

Comment:
why is this no longer using the label selector? Is the reason because the label is not accurate for coschedule = pipelineruns?

Reply:
oh sorry-- I just didn't read this carefully enough. It might be worth adding a quick comment about why it's required for isolate pipelineruns but optional for the other methods.